src/platform/nrfconnect/FactoryDataProvider.h

```diff
@@ -76,11 +76,21 @@ struct InternalFlashFactoryData
         // and make sure we do not overlap with settings partition
         constexpr size_t kFactoryDataBlockEnd =
             (FACTORY_DATA_ADDRESS + FACTORY_DATA_SIZE + CONFIG_FPROTECT_BLOCK_SIZE - 1) & (-CONFIG_FPROTECT_BLOCK_SIZE);
-        static_assert(kFactoryDataBlockEnd <= PM_SETTINGS_STORAGE_ADDRESS,
+
+        constexpr size_t kFactoryDataBlockBegin = FACTORY_DATA_ADDRESS & (-CONFIG_FPROTECT_BLOCK_SIZE);
```

Comment:
@ArekBalysNordic could you verify the alignment of the factory data beginning? TBH I don't know where this magic with & and - fprotect block size comes from.